src/flow/src/batching_mode/task.rs

```diff
@@ -0,0 +1,836 @@
+                                context: format!("Failed to rewrite plan {:?}", self.plan),
+                            })?
+                            .data;
+                        let schema_len = plan.schema().fields().len();
+
+                        // since no time window lower/upper bound is found, just return the original query(with auto columns)
+                        return Ok(Some((plan, schema_len)));
+                    }
+                }
+            };
+
+            debug!(
+                "Flow id={:?}, Generated filter expr: {:?}",
+                self.flow_id,
+                expr.as_ref()
+                    .map(|expr| expr_to_sql(expr).with_context(|_| DatafusionSnafu {
+                        context: format!("Failed to generate filter expr from {expr:?}"),
+                    }))
+                    .transpose()?
+                    .map(|s| s.to_string())
+            );
+
+            let Some(expr) = expr else {
+                // no new data, hence no need to update
+                debug!("Flow id={:?}, no new data, not update", self.flow_id);
+                return Ok(None);
+            };
+
+            let mut add_filter = AddFilterRewriter::new(expr);
+            let mut add_auto_column = AddAutoColumnRewriter::new(sink_table_meta.schema.clone());
+            // make a not optimized plan for clearer unparse
+            let plan =
+                sql_to_df_plan(query_ctx.clone(), engine.clone(), &self.query, false).await?;
+            plan.clone()
+                .rewrite(&mut add_filter)
+                .and_then(|p| p.data.rewrite(&mut add_auto_column))
+                .with_context(|_| DatafusionSnafu {
+                    context: format!("Failed to rewrite plan {plan:?}"),
+                })?
+                .data
+        };
+
+        Ok(Some((new_plan, schema_len)))
+    }
+}
+
+// auto created table have a auto added column `update_at`, and optional have a `AUTO_CREATED_PLACEHOLDER_TS_COL` column for time index placeholder if no timestamp column is specified
+// TODO(discord9): unit test
+fn create_table_with_expr(
+    plan: &LogicalPlan,
+    sink_table_name: &[String; 3],
+) -> Result<CreateTableExpr, Error> {
+    let catalog_name = &sink_table_name[0];
+    let schema_name = &sink_table_name[1];
+    let table_name = &sink_table_name[2];
+
+    let fields = plan.schema().fields();
+    let (first_time_stamp, primary_keys) = build_primary_key_constraint(plan, fields)?;
+
+    let mut column_schemas = Vec::new();
+    for field in fields {
+        let name = field.name();
+        let ty = ConcreteDataType::from_arrow_type(field.data_type());
+        let col_schema = if first_time_stamp == Some(name.clone()) {
+            ColumnSchema::new(name, ty, false).with_time_index(true)
+        } else {
+            ColumnSchema::new(name, ty, true)
+        };
+        column_schemas.push(col_schema);
+    }
+
+    let update_at_schema = ColumnSchema::new(
```

Comment:
The name is too common to confuse. Maybe we can add a prefix such as `greptime_flow_update_at` or like `AUTO_CREATED_PLACEHOLDER_TS_COL`.